contracts/src/v0.8/dev/ArbitrumValidator.sol

```diff
@@ -0,0 +1,140 @@
+  SimpleWriteAccessController private s_gasConfigAccessController;
+
+  struct GasConfiguration {
+    uint256 maximumSubmissionCost;
+    uint32 maximumGasPrice;
+    uint256 gasCostL2;
+    address refundableAddress;
+  }
+  GasConfiguration private s_gasConfig;
+  uint32 constant private s_L2GasLimit = 30000000;
+
+  /**
+   * @param aggregatorAddress default aggregator with access to validate
+   * @param inboxAddress address of the Arbitrum Inbox L1 contract
+   * @param flagAddress address of the Chainlink L2 Flags contract
+   * @param gasConfigAccessController address of the access controller for managing gas price on Arbitrum
+   * @param maxSubmissionCost maximum cost willing to pay on L2
+   * @param maximumGasPrice maximum gas price to pay on L2
+   * @param gasCostL2 value to send to L2 to cover gas fee
+   * @param refundableAddress address where gas excess on L2 will be sent
+   */
+  constructor(
+    address aggregatorAddress,
+    address inboxAddress,
+    address flagAddress,
+    address gasConfigAccessController,
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  ) {
+    require(flagAddress != address(0), "Invalid Flags contract address");
+    s_arbitrumInbox = ArbitrumInboxInterface(inboxAddress);
+    s_gasConfigAccessController = SimpleWriteAccessController(gasConfigAccessController);
+    s_flagsAddress = flagAddress;
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+
+    this.addAccess(aggregatorAddress);
+  }
+  
+  fallback() external payable {}
+
+  function withdrawFunds() 
+    external 
+    onlyOwner() 
+  {
+    address payable to = payable(msg.sender);
+    to.transfer(address(this).balance);
+  }
+
+  function withdrawFundsTo(
+    address payable to
+  ) 
+    external
+    onlyOwner() 
+  {
+    to.transfer(address(this).balance);
+  }
+
+  function setGasConfiguration(
+    uint256 maxSubmissionCost,
+    uint32 maximumGasPrice,
+    uint256 gasCostL2,
+    address refundableAddress
+  )
+    external
+  {
+    require(s_gasConfigAccessController.hasAccess(msg.sender, msg.data), "Only gas configuration admin can call");
+    _setGasConfiguration(maxSubmissionCost, maximumGasPrice, gasCostL2, refundableAddress);
+  }
+
+  function validate(
+    uint256 /* previousRoundId */,
+    int256 /* previousAnswer */,
+    uint256 /* currentRoundId */,
+    int256 currentAnswer
+  ) 
+    external 
+    override
+    checkAccess() 
+    returns (bool) 
+  {
+    // Avoids resending to L2 the same tx on every call
+    if (s_lastChangedAnswer == currentAnswer) {
+      return true;
+    }
+    s_lastChangedAnswer = currentAnswer;
+    bytes memory data = currentAnswer == 1 ? abi.encodeWithSelector(RAISE_SELECTOR, s_arbitrumFlag) : abi.encodeWithSelector(LOWER_SELECTOR, s_arbitrumFlag);
+
+    s_arbitrumInbox.createRetryableTicket{value: s_gasConfig.gasCostL2}(
```

Comment:
@RodrigoAD keep in mind that FA which we run on Optimism has a non-configurable static [VALIDATOR_GAS_LIMIT](https://github.com/smartcontractkit/chainlink/blob/55bacd4a9e07bc9121eb95c33f9e6861b9e311ea/contracts/src/v0.6/FluxAggregator.sol#L869) set, and we will probably need to increase it for this use case.

Reply:
For Arbitrum L1 Validator, we will actually set `gasCostL2` to zero, and fund the L2 `validator` as it will be paying the fee. 
`{value: s_gasConfig.gasCostL2}` this is just in case we want to pay it from L1

Reply:
This should be more explicitly documented!